File: models/datatypes/arg.cpp
#include "arg.h"


Arg::Arg(string variable_name, LVOV argument_value)
{
    this->variable_name = variable_name;
    this->argument_value = argument_value;
}


Arg::Arg(json jdata) : argument_value(jdata["argument_value"])
{
    variable_name = jdata["variable_name"];
}


string Arg::describe()
{
    return variable_name + ": " + argument_value.to_string();
}


json Arg::to_json()
{
    return {
        { "variable_name", variable_name },
        { "argument_value", argument_value.to_json() }   
    };
}

File: server/models/datatypes/arg.py
from models.datatypes.lvov import LVOV


class Arg(object):

    def __init__(self, arg_json):
        self.spec = arg_json

        self.variable_name = arg_json["variable_name"]  # str
        self.argument_value = LVOV(arg_json["argument_value"])  # lvov

